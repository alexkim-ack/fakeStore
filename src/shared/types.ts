/**
 * Product type.
 * @param category: Category product is in
 * @param description: Describes the product
 * @param id: Product id
 * @param image: Image url
 * @param price: Price of product
 * @param rating: Object containing rate, the average rating,
 * and count, the number of ratings
 * @param title: What the product is labeled as
 */
export interface ProductType {
    category: string;
    description: string;
    id: number;
    image: string;
    price: number;
    rating: { rate: number; count: number };
    title: string;
}
