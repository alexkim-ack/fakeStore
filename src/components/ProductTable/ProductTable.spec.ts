import { expect, describe, it } from "vitest";

import { ProductTable } from "./index";
import { ProductType } from "@/shared/types";

const test: ProductType[] = [
    {
        category: "men's clothing",
        description:
            "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
        id: 1,
        image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
        price: 109.95,
        rating: { rate: 3.9, count: 120 },
        title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
    },
];

describe("ProductTable", () => {
    it("should render", () => {
        const render = ProductTable({
            products: test,
            isLoading: false,
            error: null,
        });
        expect(render).toMatchSnapshot();
    });
});
