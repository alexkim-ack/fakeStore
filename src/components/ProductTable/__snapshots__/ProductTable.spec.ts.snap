// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should render 1`] = `
<ul
  className="list-none grid grid-cols-4"
>
  <li
    className="flex-auto p-5"
  >
    <img
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
    <p>
      <span>
        $109.95 
      </span>
      <span>
        Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
      </span>
    </p>
  </li>
</ul>
`;
